### PR TITLE
Reapply HMAN-578 and HMAN-846

### PR DESCRIPTION
**Pull request by RebeccaBaker**
This reverts commit 30bbc1e03944fe53b4b7935f92d464e652df5223.

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
re apply HMAN-578
https://tools.hmcts.net/jira/browse/HMAN-846

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by RebeccaBaker — COMMENTED** [review]
@padmapriyanalam can we look into the sonar issues please https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=517&id=uk.gov.hmcts.reform%3Ahmc-cft-hearing-service

**Comment by padmapriyanalam** [comment]
> @padmapriyanalam can we look into the sonar issues please https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=517&id=uk.gov.hmcts.reform%3Ahmc-cft-hearing-service



> @padmapriyanalam can we look into the sonar issues please https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=517&id=uk.gov.hmcts.reform%3Ahmc-cft-hearing-service

Addressed the sonar issues

**Review by RebeccaBaker — COMMENTED** [review]
happy with the changes

File: src/test/java/uk/gov/hmcts/reform/hmc/utils/TestingUtil.java [hunk 1]
```diff
@@ -1434,5 +1445,38 @@ public static HearingDetails hearingDetailsWithoutHearingWindow() {
...
+        linkedHearingStatusAuditEntity.setLinkedHearingEvent(CREATE_HEARING_REQUEST);
+        linkedHearingStatusAuditEntity.setHttpStatus(String.valueOf(HttpStatus.SC_OK));
+        linkedHearingStatusAuditEntity.setSource(HMC);
+        linkedHearingStatusAuditEntity.setTarget(HMI);
+        JsonNode linkedGroupHearings = OBJECT_MAPPER.convertValue("2000000000", JsonNode.class);
+        linkedHearingStatusAuditEntity.setLinkedGroupHearings(linkedGroupHearings);
+        return linkedHearingStatusAuditEntity;
+    }
+
+    private static final ObjectMapper OBJECT_MAPPER = new Jackson2ObjectMapperBuilder()
```
Comment (markdathornehmcts) [inline src/test/java/uk/gov/hmcts/reform/hmc/utils/TestingUtil.java:1477]: Not sure I like the specific reference to Jdk8. CoPilot suggests:

You can streamline the creation of an `ObjectMapper` instance with default settings, including registering modules for JDK 8 types like `Optional`, by directly instantiating `ObjectMapper` and then using its `findAndRegisterModules` method. This method automatically discovers and registers all modules on the classpath, including the `Jdk8Module`, without explicitly listing them. This approach reduces boilerplate code and ensures that your `ObjectMapper` is configured with all available modules, improving maintainability and future-proofing your serialization and deserialization logic.

Here's how you can do it:

```java
private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().findAndRegisterModules();
```

**Review by markdathornehmcts — APPROVED** [review]
LGTM bar the comment suggested. Please check.

**Pull request merged**